crates/libs/core/src/types/client/health.rs

```diff
@@ -0,0 +1,947 @@
+            ServiceManifestName: value.service_manifest_name.as_pcwstr(),
+            NodeName: value.node_name.as_pcwstr(),
+            HealthInformation: &FABRIC_HEALTH_INFORMATION::from(&value.health_information),
+            Reserved: std::ptr::null_mut(),
+        }
+    }
+}
+
+impl From<&FABRIC_DEPLOYED_SERVICE_PACKAGE_HEALTH_REPORT> for DeployedServicePackageHealthReport {
+    fn from(value: &FABRIC_DEPLOYED_SERVICE_PACKAGE_HEALTH_REPORT) -> Self {
+        Self {
+            application_name: WString::from_wide(unsafe {
+                PCWSTR::from_raw(value.ApplicationName.0).as_wide()
+            }),
+            service_manifest_name: WString::from_wide(unsafe {
+                value.ServiceManifestName.as_wide()
+            }),
+            node_name: WString::from_wide(unsafe { value.NodeName.as_wide() }),
+            health_information: HealthInformation::from(unsafe {
+                value.HealthInformation.as_ref().unwrap()
+            }),
+        }
+    }
+}
+
+/// FABRIC_CLUSTER_HEALTH_REPORT
+#[derive(Debug, Clone)]
+pub struct ClusterHealthReport {
+    pub health_information: HealthInformation,
+    // TODO: Implement reserved fields
+    // pub reserved: *mut std::ffi::c_void,
+}
+
+impl From<&ClusterHealthReport> for FABRIC_CLUSTER_HEALTH_REPORT {
+    fn from(value: &ClusterHealthReport) -> Self {
+        Self {
+            HealthInformation: &FABRIC_HEALTH_INFORMATION::from(&value.health_information),
+            Reserved: std::ptr::null_mut(),
+        }
+    }
+}
+
+impl From<&FABRIC_CLUSTER_HEALTH_REPORT> for ClusterHealthReport {
+    fn from(value: &FABRIC_CLUSTER_HEALTH_REPORT) -> Self {
+        Self {
+            health_information: HealthInformation::from(unsafe {
+                value.HealthInformation.as_ref().unwrap()
+            }),
+        }
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use super::*;
+
+    #[test]
+    fn test_cluster_health_report_conversion() {
+        let health_info = HealthInformation {
+            source_id: "source_id".into(),
+            property: "property".into(),
+            time_to_live_seconds: 10,
+            state: crate::types::HealthState::Ok,
+            description: "description".into(),
+            sequence_number: 1,
+            remove_when_expired: false,
+        };
+
+        let cluster_health_report = HealthReport::Cluster(ClusterHealthReport {
+            health_information: health_info.clone(),
+        });
+
+        let com_cluster_health_report: FABRIC_HEALTH_REPORT = (&cluster_health_report).into();
+        assert_eq!(
+            com_cluster_health_report.Kind,
+            FABRIC_HEALTH_REPORT_KIND_CLUSTER
+        );
+
+        let cluster_health_report2: HealthReport = (&com_cluster_health_report).into();
+        matches!(cluster_health_report2, HealthReport::Cluster(_));
+
+        // Check the inner values are matching
+        if let HealthReport::Cluster(report) = cluster_health_report2 {
```

Comment:
let HealthReport::Cluster(report) = cluster_health_report2  else { panic!() } and remove the above matches!.

Reply:
Done

Reply:
U can remove the if and save the indentation.

Reply:
Addressed.